### PR TITLE
WIP: testing dependent-destroy

### PR DESCRIPTION
**Pull request by jameswnl**
See the different behaviors of the 2 added specs.

### CODE REVIEW

**Comment by Fryguy** [comment]
I just don't get it...need to investigate more :/

**Comment by Ladas** [comment]
the data are not stored properly

check 
```
subject.automation_manager.reload.configured_systems
```

D, [2018-01-10T21:22:01.373714 #12596] DEBUG -- :   ManageIQ::Providers::AutomationManager::ConfiguredSystem Load (0.3ms)  SELECT "configured_systems".* FROM "configured_systems" WHERE "configured_systems"."type" IN ('ManageIQ::Providers::AutomationManager::ConfiguredSystem', 'ManageIQ::Providers::ExternalAutomationManager::ConfiguredSystem', 'ManageIQ::Providers::EmbeddedAutomationManager::ConfiguredSystem', 'ManageIQ::Providers::AnsibleTower::AutomationManager::ConfiguredSystem', 'ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfiguredSystem') AND "configured_systems"."manager_id" = $1  [["manager_id", 2952]]

The configured system needs to have valid STI type, otherwise the query will not find it. subject.automation_manager.configured_systems is not doing any query, it just shows cached data

**Comment by jameswnl** [comment]
@Ladas see my fix here https://github.com/ManageIQ/manageiq/pull/16798

File: spec/support/ansible_shared/provider.rb [hunk 1]
```diff
@@ -36,6 +36,46 @@
       expect(OperatingSystem.count).to       eq(0)
       expect(Hardware.count).to              eq(0)
     end
+
+    it "thru automation_manager will remove all child objects" do
+      subject.automation_manager.configured_systems = [
+        FactoryGirl.create(:configured_system, :computer_system =>
```
Comment (Ladas) [inline spec/support/ansible_shared/provider.rb]: you should use different factory https://github.com/Ladas/manageiq/blob/2f0ef1a90758f2b5c37e93a0ed2f586a916dad64/spec/factories/configured_system.rb#L10

or the default factory should contain valid base class STI, in this case ManageIQ::Providers::AutomationManager::ConfiguredSystem

-----

None of factories should be producing a record that can't be created in real life, e.g. ConfiguredSystem with type nil is not valid, refresh always set :type

**Comment by jameswnl** [comment]
thanks to @chrisarcand problem solved, closing this

File: spec/support/ansible_shared/provider.rb (hunk 1 above)
Comment (chrisarcand) [reply spec/support/ansible_shared/provider.rb]: Thanks @Ladas !

**Pull request closed without merging**